### PR TITLE
Compatible with old cmake

### PR DESCRIPTION
**Pull request by romainreignier**
Fix https://github.com/ethz-adrl/control-toolbox/pull/116#issuecomment-625800528 after discussion in https://github.com/ethz-adrl/control-toolbox/pull/129

I have tested it on Xenial and Bionic but because ROS is not yet released on Focal, I did not test the case when Cmake > 3.12. @acxz could you try it please?

I have also fixed the crash on the `hist` test of `plotTest`.

### CODE REVIEW

**Review by markusgft — APPROVED** [review]
Looks very good to me, although I am not able to test this in my current PC setup. 
@acxz can you confirm that it covers your usecase, too?
If everyone is happy, please feel free to trigger the merge.

**Comment by acxz** [comment]
I'm sorry to report that it does not work on my system. Let me dig a bit deeper.

File: ct_core/CMakeLists.txt [hunk 1]
```diff
@@ -63,18 +63,30 @@ else()
...
+        list(APPEND ct_core_COMPILE_DEFINITIONS PLOTTING_ENABLED)
+        set(PLOTTING_ENABLED ON)
+        message(STATUS "Python library path ... " ${PYTHON_LIBRARY})
+        list(APPEND ct_core_LIBS ${PYTHON_LIBRARY})
+    else()
+        message(STATUS "Python 3 not found, plotting will not be available")
+    endif()
 else()
-   message(STATUS "Python 3 not found, plotting will not be available")
+    find_package(Python COMPONENTS Development Numpy QUIET)
```
Comment (acxz) [inline ct_core/CMakeLists.txt]:
```suggestion
    find_package(Python COMPONENTS Development NumPy QUIET)
```

Just a small typo

**Review by acxz — CHANGES_REQUESTED** [review]

File: ct_core/CMakeLists.txt (hunk 1 above)
Comment (acxz) [reply ct_core/CMakeLists.txt]: PR is ready with this fix

**Comment by romainreignier** [comment]
Ah sorry, my mistake. I will fix it tomorrow.

22 mai 2020 23:30:42 acxz <notifications@github.com>:

**Comment by romainreignier** [comment]
Thank you @markusgft for fixing the typo.
I was away from home this week-end.

Anyway, I have managed to test it on Ubuntu Focal and ROS [Noetic testing](https://discourse.ros.org/t/ros-noetic-call-for-testing/14203).
It works, the histogram is displayed in `plotTest` but a seg fault happen on the next test while it was working on Ubuntu 18.04 and 16.04.
@acxz Does the `plotTest` seg fault on your side? After the histogram?

**Comment by romainreignier** [comment]
From a quick `gdb` session, the segfault seems to be in the histogram `show()` call, even if it is already shown on the screen.
But that is another issue, not related at all with CMake version.

**Comment by acxz** [comment]
Yep can reproduce, I have a pretty big hunch that if we incorporate upstream changes from (`matplotlib-cpp`)[https://github.com/lava/matplotlib-cpp] specifically this commit: https://github.com/lava/matplotlib-cpp/commit/60dcd64c8fd4766e5426f57decfb765422a1d3fe it'll fix it.

This is the particular line that prob needs to be changed: https://github.com/lava/matplotlib-cpp/commit/60dcd64c8fd4766e5426f57decfb765422a1d3fe#diff-59dbbec4311029188f570e995bb6a594R714 in our local copy of the plotting script.

**Comment by romainreignier** [comment]
Interesting, but I have try to add a couple of them and it does not fix anything: https://github.com/romainreignier/control-toolbox/commit/f2349698ba9c4295a266a3fe08c1f6e4ed69de47

**Pull request merged**